hummingbot/client/command/config_command.py

```diff
@@ -194,21 +223,107 @@ async def _config_single_key(self,  # type: HummingbotApplication
+        config_var = config_map[key]
+        if input_value is None:
+            self._notify("Please follow the prompt to complete configurations: ")
+        if config_var.key == "inventory_target_base_pct":
+            await self.asset_ratio_maintenance_prompt_legacy(config_map, input_value)
+        elif config_var.key == "inventory_price":
+            await self.inventory_price_prompt_legacy(config_map, input_value)
+        else:
+            await self.prompt_a_config_legacy(config_var, input_value=input_value, assign_default=False)
+        if self.app.to_stop_config:
+            self.app.to_stop_config = False
+            return
+        await self.update_all_secure_configs_legacy()
+        missings = missing_required_configs_legacy(config_map)
+        if missings:
+            self._notify("\nThere are other configuration required, please follow the prompt to complete them.")
+        missings = await self._prompt_missing_configs(config_map)
+        save_to_yml_legacy(file_path, config_map)
+        self._notify("\nNew configuration saved:")
+        self._notify(f"{key}: {str(config_var.value)}")
+        self.app.app.style = load_style()
+        for config in missings:
+            self._notify(f"{config.key}: {str(config.value)}")
+        if (
+            isinstance(self.strategy, PureMarketMakingStrategy) or
+            isinstance(self.strategy, PerpetualMarketMakingStrategy)
+        ):
+            updated = ConfigCommand.update_running_mm(self.strategy, key, config_var.value)
+            if updated:
+                self._notify(f"\nThe current {self.strategy_name} strategy has been updated "
+                             f"to reflect the new configuration.")
+
     async def _prompt_missing_configs(self,  # type: HummingbotApplication
                                       config_map):
-        missings = missing_required_configs(config_map)
+        missings = missing_required_configs_legacy(config_map)
         for config in missings:
-            await self.prompt_a_config(config)
+            await self.prompt_a_config_legacy(config)
             if self.app.to_stop_config:
                 self.app.to_stop_config = False
                 return
-        if missing_required_configs(config_map):
+        if missing_required_configs_legacy(config_map):
             return missings + (await self._prompt_missing_configs(config_map))
         return missings
 
-    async def asset_ratio_maintenance_prompt(self,  # type: HummingbotApplication
-                                             config_map,
-                                             input_value = None):
+    async def asset_ratio_maintenance_prompt(
+        self,  # type: HummingbotApplication
+        config_map: BaseTradingStrategyConfigMap,
+        input_value: Any = None,
+    ):  # pragma: no cover
+        """
+        Will be removed: https://app.shortcut.com/coinalpha/story/24923/collect-special-case-prompts-via-config-models
```

Comment:
I am thinking that this will be removed before we merge the feature branch — hoping that we can add the task to the epic — but will remove it in case.